### PR TITLE
Catch font smoothing on all body elements

### PR DESCRIPTION
**Pull request by acconrad**
`-webkit-font-smoothing` will not catch everything in `body`, this can be verified by creating a `<button>` on a screen next to a `<a class="button">` and noting the difference in font smoothing. This PR will fix font smoothing and keep it consistent across all HTML elements.

### CODE REVIEW

**Comment by tysongach** [comment]
Is it only the `button` that doesn’t inherit this, or is this a more widespread issue? I only ask because we want to try to reduce wildcard selectors whenever possible…

**Comment by joshuaogle** [comment]
This is kind of a tricky one, and is either a bug in Chrome or expected behavior, depending on who you ask. There are a few form elements that don't inherit things like font-smoothing, font-size, and a few other styles: `button`, `input`, `textarea` and `select`.

Text inputs will inherit if you inherit it like this:
```
#{all-text-inputs},
textarea {
  -webkit-font-smoothing: inherit;
  font-family: inherit;
}
```

You can also get buttons, input[type="submit"] and selects to inherit like this, but you have to add in `-webkit-appearance: none;`, which I often do in my own projects but I wouldn't want to add to Bitters.

So I wouldn't call this a bug in Bitters, because this gets into the sticky business of overriding browser behavior, but hopefully that helps you solve the problem in your project.

**Comment by acconrad** [comment]
My only concern is that while it may not be a Bitter-specific bug, it does create inconsistencies in font-rendering behavior, which the end user may not be aware of. They shouldn't expect that if they create a link button and a button button next to each other, that they should look different. It should *just work*. To that end, either this PR should be merged or the suggestion by @joshuaogle should be implemented.

**Comment by whmii** [comment]
I agree with @tysongach that we reeeeeealy want to avoid wildcard selectors and font-smoothing just isn;t important enough to make an exception to that rule. I would be down for this if it was the following…

```
#{all-text-inputs},
textarea {
  -webkit-font-smoothing: inherit;
  font-family: inherit;
}
```

File: app/assets/stylesheets/_typography.scss [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-  -webkit-font-smoothing: antialiased;
   background-color: $base-background-color;
   color: $base-font-color;
   font-family: $base-font-family;
   font-size: $base-font-size;
   line-height: $base-line-height;
+  -webkit-font-smoothing: antialiased;
+  
+  #{$all-text-inputs},
+  textarea {
```
Comment (tysongach) [inline app/assets/stylesheets/_typography.scss]: I think these styles should go inside the same ruleset found in [`_forms.scss`](https://github.com/thoughtbot/bitters/blob/master/app/assets/stylesheets/_forms.scss).

File: app/assets/stylesheets/_typography.scss [hunk 2]
```diff
@@ -1,11 +1,17 @@
 body {
   @include font-feature-settings("kern","liga","frac", "pnum");
-  -webkit-font-smoothing: antialiased;
   background-color: $base-background-color;
   color: $base-font-color;
   font-family: $base-font-family;
   font-size: $base-font-size;
   line-height: $base-line-height;
+  -webkit-font-smoothing: antialiased;
```
Comment (tysongach) [inline app/assets/stylesheets/_typography.scss]: :abc: This should be sorted just below `font-size`.

File: app/assets/stylesheets/_typography.scss (hunk 1 above)
Comment (tysongach) [inline app/assets/stylesheets/_typography.scss]: @acconrad Any update here?

File: app/assets/stylesheets/_typography.scss (hunk 2 above)
Comment (tysongach) [inline app/assets/stylesheets/_typography.scss]: @acconrad Any update here?

**Comment by tysongach** [comment]
@acconrad Still a few open comments and questions here (see collapsed comments above). Also, can you please rebase against `master` to pull in the latest changes?

**Comment by acconrad** [comment]
@tysongach what open comments are there? I moved the font smoothing into the forms file and refactored the typography file to be in the correct attribute order.

**Comment by tysongach** [comment]
@acconrad Couple of questions:

1. Your original comment mentions the `button` element, but all changes in this PR are on text inputs (via Bourbon’s [`#{$all-text-inputs}`](http://bourbon.io/docs/#text-inputs) helper). Are you certain these code changes are what you’re expecting?
2. Can you [update your fork](https://robots.thoughtbot.com/keeping-a-github-fork-updated) and rebase against `master`? This PR is out of date with the latest changes to Bitters.
3. Can you post a screenshot of the font smoothing acting differently across certain elements? I’m not sure I’m finding the same thing you are. Here’s some dummy code (styled with a fresh Bitters instance) and a screenshot from my retina MacBook Pro (Chrome on the left, Safari on the right):

```html
<button>Submit</button>
<input type="submit" value="Submit">
<a class="button">Submit</a>
```

buttons

I don’t see any difference between the three element types, nor across browsers.

**Comment by acconrad** [comment]
It appears as if the current webkit browsers no longer have this issue - it must have been only relevant back in December. Closing this unless I find it again

**Pull request closed without merging**